engines/osxnative/sandbox.go

```diff
@@ -0,0 +1,68 @@
+package macengine
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"io"
+	"os"
+	"os/exec"
+)
+
+type sandbox struct {
+	context     *runtime.TaskContext
+	taskPayload *payload
+	env         *[]string
+	aborted     bool
+}
+
+func newSandbox(context *runtime.TaskContext, taskPayload *payload, env *[]string) *sandbox {
+	return &sandbox{
+		context,
+		taskPayload,
+		env,
+		false,
+	}
+}
+
+func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
+	if s.aborted {
+		return resultset{success: false}, engines.ErrSandboxAborted
+	}
+
+	// Use the host environment plus custom environment variables
+	env := os.Environ()
```

Comment:
I think that we should not copy any host environment variables if we can.  If we do need some of the host variables, we can white list those instead.  Secrets could be in the host variables.